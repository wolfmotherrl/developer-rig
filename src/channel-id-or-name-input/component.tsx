import * as React from 'react';
import { DeveloperRigUserId } from '../constants/rig';

interface Props {
  isUser?: boolean;
  name: string;
  value: string;
  labelClassName?: string;
  textClassName?: string;
  onChange: (event: React.FormEvent<HTMLInputElement>) => void;
}

interface State {
  checked: boolean;
  lastChannelId: string;
}


export class ChannelIdOrNameInput extends React.Component<Props, State> {
  state: State = {
    checked: this.props.value === DeveloperRigUserId,
    lastChannelId: '',
  }

  public render() {
    return (
      <>
        <label className={this.props.labelClassName}>
          <span className={this.props.textClassName}>
            {this.props.isUser ? 'User' : 'Channel'} ID or Name
          </span>
          <input type="text" name={this.props.name} value={this.props.value} onChange={this.props.onChange} disabled={this.state.checked} />
        </label>
        <label className={this.props.labelClassName}>
          <input type="checkbox" name="wantsDeveloperRigUserId" onChange={this.onChange} checked={this.state.checked} />
          Use Developer Rig user
        </label>
      </>
    );
  }

  private onChange = (event: React.FormEvent<HTMLInputElement>) => {
    const { checked } = event.currentTarget;
    if (checked) {
      this.setState({ lastChannelId: this.props.value });
      this.props.onChange({ currentTarget: { name: this.props.name, value: DeveloperRigUserId } } as React.FormEvent<HTMLInputElement>);
    } else {
      this.props.onChange({ currentTarget: { name: this.props.name, value: this.state.lastChannelId } } as React.FormEvent<HTMLInputElement>);
    }
    this.setState({ checked });
  }
}
