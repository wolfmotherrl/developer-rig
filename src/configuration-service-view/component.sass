.configuration-service-view
  display: grid
  grid-template-columns: 1fr 8pt 1fr
  grid-gap: 8pt
  margin-top: 8pt

  &__button
    background-color: #6441A4
    border: 0
    color: white
    height: 15pt
    margin-left: 8pt
    min-width: 8em

    &--first
      margin-left: 0

  &__title
    font-size: larger
    font-weight: bold

  &-property
    display: block
    margin: 1em

    &__input
      width: 70%

      &--error
        border-color: red

    &__name
      margin-bottom: 5px
      margin-right: 11px

    &__value
      color: grey
      user-select: text

  &__vertical-bar
    background-color: #DAD8DE
    width: 1pt;
    height: 100%

  &__item
    margin-bottom: 8pt
    margin-left: -17pt

  &__item-text
    margin: 3pt

    &--title
      font-weight: bold
