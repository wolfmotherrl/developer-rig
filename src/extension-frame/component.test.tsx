import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { setupMountTest } from '../tests/enzyme-util/mount';
import { createExtensionForTest } from '../tests/constants/extension';
import { ExtensionFrame } from './component';
import { ExtensionViewType, ExtensionAnchor, ExtensionMode } from '../constants/extension-coordinator';

const defaultPropGenerator = () => ({
  channelId: 'twitch',
  className: 'view',
  configuration: {},
  frameId: '0',
  installationAbilities: {
    isChatEnabled: true,
  },
  extension: createExtensionForTest(),
  type: ExtensionAnchor.Panel,
  mode: ExtensionMode.Viewer,
  iframe: '',
  isLocal: false,
  isPopout: false,
  bindIframeToParent: jest.fn(),
});

const setupShallow = setupShallowTest(ExtensionFrame, defaultPropGenerator);
const setupMount = setupMountTest(ExtensionFrame, defaultPropGenerator);

describe('<ExtensionFrame />', () => {
  it('local onload postMessages data correctly', () => {
    const { wrapper } = setupMount({ isLocal: true });
    const mockIframeRef: any = {
      contentWindow: {
        postMessage: jest.fn(),
      },
    };
    const instance = wrapper.instance() as ExtensionFrame;
    instance.iframe = mockIframeRef;
    instance.extensionFrameInit();
    expect(mockIframeRef.contentWindow.postMessage).toHaveBeenCalledWith(createExpected('panel', true), '*');
  });

  it('online onload postMessages data correctly', () => {
    const { wrapper } = setupMount();
    const mockIframeRef: any = {
      contentWindow: {
        postMessage: jest.fn(),
      },
    };
    const instance = wrapper.instance() as ExtensionFrame;
    instance.iframe = mockIframeRef;
    instance.extensionFrameInit();
    expect(mockIframeRef.contentWindow.postMessage).toHaveBeenCalledWith(createExpected('panel'), '*');
  });

  it('local onload postMessages data correctly when platform is mobile', () => {
    const { wrapper } = setupMount({
      isLocal: true,
      type: ExtensionViewType.Mobile,
    });
    const mockIframeRef: any = {
      contentWindow: {
        postMessage: jest.fn(),
      },
    };
    const instance = wrapper.instance() as ExtensionFrame;
    instance.iframe = mockIframeRef;
    instance.extensionFrameInit();
    expect(mockIframeRef.contentWindow.postMessage).toHaveBeenCalledWith(createExpected('mobile', true), '*');
  });

  it('online onload postMessages data correctly when platform is mobile', () => {
    const { wrapper } = setupMount({
      type: ExtensionViewType.Mobile,
    });
    const mockIframeRef: any = {
      contentWindow: {
        postMessage: jest.fn(),
      },
    };
    const instance = wrapper.instance() as ExtensionFrame;
    instance.iframe = mockIframeRef;
    instance.extensionFrameInit();
    expect(mockIframeRef.contentWindow.postMessage).toHaveBeenCalledWith(createExpected('mobile'), '*');
  });

  describe('when in local live config mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        isLocal: true,
        mode: ExtensionMode.Dashboard,
        type: ExtensionViewType.LiveConfig,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in online live config mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        mode: ExtensionMode.Dashboard,
        type: ExtensionViewType.LiveConfig,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in local config mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        isLocal: true,
        mode: ExtensionMode.Config,
        type: ExtensionViewType.Config,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in online config mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        mode: ExtensionMode.Config,
        type: ExtensionViewType.Config,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in local panel mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        isLocal: true,
        mode: ExtensionMode.Viewer,
        type: ExtensionAnchor.Panel,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in online panel mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        mode: ExtensionMode.Viewer,
        type: ExtensionAnchor.Panel,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in local video overlay mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        isLocal: true,
        mode: ExtensionMode.Viewer,
        type: ExtensionAnchor.Overlay,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in online video overlay mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        mode: ExtensionMode.Viewer,
        type: ExtensionAnchor.Overlay,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  function createExpected(anchor: string, isLocal?: boolean) {
    const expected = {
      action: 'extension-frame-init',
      channelId: 'twitch',
      parameters: {
        anchor: anchor,
        channelId: NaN,
        configuration: {},
        extension: {
          authorName: 'test',
          bitsEnabled: false,
          clientId: 'mockClientId',
          description: 'description',
          hasChatSupport: false,
          iconUrl: 'iconUrl',
          iconUrls: { square100: '100x100' },
          id: 'id',
          name: 'name',
          requestIdentityLink: false,
          sku: 'sku',
          state: 'Released',
          summary: 'summary',
          token: 'token',
          vendorCode: 'vendorCode',
          version: '0.1',
          views: {
            config: {
              canLinkExternalContent: false,
              viewerUrl: 'https://test:8080',
            },
            component: {
              aspectHeight: 3000,
              aspectWidth: 2500,
              canLinkExternalContent: false,
              viewerUrl: 'https://test:8080',
              zoom: false,
              zoomPixels: 100,
            },
            liveConfig: {
              canLinkExternalContent: false,
              viewerUrl: 'https://test:8080',
            },
            mobile: {
              viewerUrl: 'https://test:8080',
            },
            panel: {
              canLinkExternalContent: false,
              height: 300,
              viewerUrl: 'https://test:8080',
            },
            videoOverlay: {
              canLinkExternalContent: false,
              viewerUrl: 'https://test:8080',
            },
          },
          whitelistedConfigUrls: ['foo'],
          whitelistedPanelUrls: ['bar'],
        },
        iframeClassName: 'extension-frame',
        installationAbilities: {
          isChatEnabled: true,
        },
        isPopout: false,
        loginId: null as null,
        mode: 'viewer',
        platform: anchor === 'mobile' ? 'mobile' : 'web',
        trackingProperties: {},
      },
      frameId: '0',
    };
    if (isLocal) {
      ['config', 'component', 'liveConfig', 'mobile', 'panel', 'videoOverlay'].forEach((viewName) => {
        const views: { [viewName: string]: { viewerUrl: string } } = expected.parameters.extension.views;
        views[viewName].viewerUrl += '?developer_rig=local';
      });
    }
    return expected;
  }
});
