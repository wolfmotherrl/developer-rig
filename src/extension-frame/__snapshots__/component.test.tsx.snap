// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExtensionFrame /> when in local config mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    channelId="twitch"
    className="view"
    configuration={Object {}}
    extension={
        Object {
            "authorName": "test",
            "bitsEnabled": false,
            "clientId": "mockClientId",
            "description": "description",
            "hasChatSupport": false,
            "iconUrl": "iconUrl",
            "iconUrls": Object {
              "square100": "100x100",
            },
            "id": "id",
            "name": "name",
            "requestIdentityLink": false,
            "sku": "sku",
            "state": "Released",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
                "zoom": false,
                "zoomPixels": 100,
              },
              "config": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "liveConfig": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "mobile": Object {
                "viewerUrl": "https://test:8080",
              },
              "panel": Object {
                "canLinkExternalContent": false,
                "height": 300,
                "viewerUrl": "https://test:8080",
              },
              "videoOverlay": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    installationAbilities={
        Object {
            "isChatEnabled": true,
          }
    }
    isLocal={true}
    isPopout={false}
    mode="config"
    type="config"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in local live config mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    channelId="twitch"
    className="view"
    configuration={Object {}}
    extension={
        Object {
            "authorName": "test",
            "bitsEnabled": false,
            "clientId": "mockClientId",
            "description": "description",
            "hasChatSupport": false,
            "iconUrl": "iconUrl",
            "iconUrls": Object {
              "square100": "100x100",
            },
            "id": "id",
            "name": "name",
            "requestIdentityLink": false,
            "sku": "sku",
            "state": "Released",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
                "zoom": false,
                "zoomPixels": 100,
              },
              "config": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "liveConfig": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "mobile": Object {
                "viewerUrl": "https://test:8080",
              },
              "panel": Object {
                "canLinkExternalContent": false,
                "height": 300,
                "viewerUrl": "https://test:8080",
              },
              "videoOverlay": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    installationAbilities={
        Object {
            "isChatEnabled": true,
          }
    }
    isLocal={true}
    isPopout={false}
    mode="dashboard"
    type="liveConfig"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in local panel mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    channelId="twitch"
    className="view"
    configuration={Object {}}
    extension={
        Object {
            "authorName": "test",
            "bitsEnabled": false,
            "clientId": "mockClientId",
            "description": "description",
            "hasChatSupport": false,
            "iconUrl": "iconUrl",
            "iconUrls": Object {
              "square100": "100x100",
            },
            "id": "id",
            "name": "name",
            "requestIdentityLink": false,
            "sku": "sku",
            "state": "Released",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
                "zoom": false,
                "zoomPixels": 100,
              },
              "config": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "liveConfig": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "mobile": Object {
                "viewerUrl": "https://test:8080",
              },
              "panel": Object {
                "canLinkExternalContent": false,
                "height": 300,
                "viewerUrl": "https://test:8080",
              },
              "videoOverlay": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    installationAbilities={
        Object {
            "isChatEnabled": true,
          }
    }
    isLocal={true}
    isPopout={false}
    mode="viewer"
    type="panel"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in local video overlay mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    channelId="twitch"
    className="view"
    configuration={Object {}}
    extension={
        Object {
            "authorName": "test",
            "bitsEnabled": false,
            "clientId": "mockClientId",
            "description": "description",
            "hasChatSupport": false,
            "iconUrl": "iconUrl",
            "iconUrls": Object {
              "square100": "100x100",
            },
            "id": "id",
            "name": "name",
            "requestIdentityLink": false,
            "sku": "sku",
            "state": "Released",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
                "zoom": false,
                "zoomPixels": 100,
              },
              "config": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "liveConfig": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "mobile": Object {
                "viewerUrl": "https://test:8080",
              },
              "panel": Object {
                "canLinkExternalContent": false,
                "height": 300,
                "viewerUrl": "https://test:8080",
              },
              "videoOverlay": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    installationAbilities={
        Object {
            "isChatEnabled": true,
          }
    }
    isLocal={true}
    isPopout={false}
    mode="viewer"
    type="video_overlay"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in online config mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    channelId="twitch"
    className="view"
    configuration={Object {}}
    extension={
        Object {
            "authorName": "test",
            "bitsEnabled": false,
            "clientId": "mockClientId",
            "description": "description",
            "hasChatSupport": false,
            "iconUrl": "iconUrl",
            "iconUrls": Object {
              "square100": "100x100",
            },
            "id": "id",
            "name": "name",
            "requestIdentityLink": false,
            "sku": "sku",
            "state": "Released",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
                "zoom": false,
                "zoomPixels": 100,
              },
              "config": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "liveConfig": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "mobile": Object {
                "viewerUrl": "https://test:8080",
              },
              "panel": Object {
                "canLinkExternalContent": false,
                "height": 300,
                "viewerUrl": "https://test:8080",
              },
              "videoOverlay": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    installationAbilities={
        Object {
            "isChatEnabled": true,
          }
    }
    isLocal={false}
    isPopout={false}
    mode="config"
    type="config"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in online live config mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    channelId="twitch"
    className="view"
    configuration={Object {}}
    extension={
        Object {
            "authorName": "test",
            "bitsEnabled": false,
            "clientId": "mockClientId",
            "description": "description",
            "hasChatSupport": false,
            "iconUrl": "iconUrl",
            "iconUrls": Object {
              "square100": "100x100",
            },
            "id": "id",
            "name": "name",
            "requestIdentityLink": false,
            "sku": "sku",
            "state": "Released",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
                "zoom": false,
                "zoomPixels": 100,
              },
              "config": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "liveConfig": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "mobile": Object {
                "viewerUrl": "https://test:8080",
              },
              "panel": Object {
                "canLinkExternalContent": false,
                "height": 300,
                "viewerUrl": "https://test:8080",
              },
              "videoOverlay": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    installationAbilities={
        Object {
            "isChatEnabled": true,
          }
    }
    isLocal={false}
    isPopout={false}
    mode="dashboard"
    type="liveConfig"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in online panel mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    channelId="twitch"
    className="view"
    configuration={Object {}}
    extension={
        Object {
            "authorName": "test",
            "bitsEnabled": false,
            "clientId": "mockClientId",
            "description": "description",
            "hasChatSupport": false,
            "iconUrl": "iconUrl",
            "iconUrls": Object {
              "square100": "100x100",
            },
            "id": "id",
            "name": "name",
            "requestIdentityLink": false,
            "sku": "sku",
            "state": "Released",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
                "zoom": false,
                "zoomPixels": 100,
              },
              "config": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "liveConfig": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "mobile": Object {
                "viewerUrl": "https://test:8080",
              },
              "panel": Object {
                "canLinkExternalContent": false,
                "height": 300,
                "viewerUrl": "https://test:8080",
              },
              "videoOverlay": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    installationAbilities={
        Object {
            "isChatEnabled": true,
          }
    }
    isLocal={false}
    isPopout={false}
    mode="viewer"
    type="panel"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in online video overlay mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    channelId="twitch"
    className="view"
    configuration={Object {}}
    extension={
        Object {
            "authorName": "test",
            "bitsEnabled": false,
            "clientId": "mockClientId",
            "description": "description",
            "hasChatSupport": false,
            "iconUrl": "iconUrl",
            "iconUrls": Object {
              "square100": "100x100",
            },
            "id": "id",
            "name": "name",
            "requestIdentityLink": false,
            "sku": "sku",
            "state": "Released",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
                "zoom": false,
                "zoomPixels": 100,
              },
              "config": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "liveConfig": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "mobile": Object {
                "viewerUrl": "https://test:8080",
              },
              "panel": Object {
                "canLinkExternalContent": false,
                "height": 300,
                "viewerUrl": "https://test:8080",
              },
              "videoOverlay": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    installationAbilities={
        Object {
            "isChatEnabled": true,
          }
    }
    isLocal={false}
    isPopout={false}
    mode="viewer"
    type="video_overlay"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
